memory/src/main/scala/filodb.memory/format/vectors/DeltaDeltaVector.scala

```diff
@@ -211,12 +211,16 @@ object DeltaDeltaDataReader extends LongVectorDataReader {
     if (slope(vector) == 0)
       return 0
     require(start >= 0 && end < length(vector), s"($start, $end) is out of bounds, length=${length(vector)}")
+    val itr = iterate(vector, start)
+    var prev: Long = 0
     var changes = 0
-    var i = start
-    while (i <= end) {
-      val current = iterate(vector, i).next
-      changes += 1
-      i = i + 1
+    for {i <- start until end} {
```

Comment:
Use optimized here, it will inline the function and make this a while loop underneath for better performance

Reply:
Added optimized